### DISCUSSION TITLE
Is there an easier way to put different hero images on different pages?

### DISCUSSION THREAD

**Discussion by toomanybrians** [squidfunk/mkdocs-material#4462] [Q&A] [answered]
Hello!

I've been playing with the hero block functionality. So far I like it, and I'm able to get it to do what I want, which is great! My question is whether there's a better way to be doing it.

I only have a few pages where I want hero images, but with a different hero image for each page. The only way I've figured out how to implement the hero image is to use a custom page template for each markdown page where I want a hero image, and then to make separate templates for each page.

This all works fine, but I wonder if there's a way I can specify which hero image is used directly from the markdown page versus having to create a separate html template for each?

Thanks!
Brian

**Comment by squidfunk** [comment, marked as answer]
You could define the hero image [via front matter](https://squidfunk.github.io/mkdocs-material/reference/#usage). You could then use that variable, e.g. `page.meta.hero` in your template:

```
---
hero: path/to/hero/image.png
---

# Document title
```

In your template:

``` html
<img src="{{ page.meta.hero }}" />
```

**Reply by toomanybrians** [reply]
awesome, thanks!!
